src/prefect/environments/execution/k8s/job.py

```diff
@@ -163,26 +149,34 @@ def run_flow(self) -> None:
                 get_default_executor_class,
             )
 
-            # Load serialized flow from file and run it with the executor
-            with open(
-                prefect.context.get(
-                    "flow_file_path", "/root/.prefect/flow_env.prefect"
-                ),
-                "rb",
-            ) as f:
-                flow = cloudpickle.load(f)
-
-                ## populate global secrets
-                secrets = prefect.context.get("secrets", {})
-                for secret in flow.storage.secrets:
-                    secrets[secret] = prefect.tasks.secrets.PrefectSecret(
-                        name=secret
-                    ).run()
-
-                with prefect.context(secrets=secrets):
-                    runner_cls = get_default_flow_runner_class()
-                    executor_cls = get_default_executor_class()(**self.executor_kwargs)
-                    runner_cls(flow=flow).run(executor=executor_cls)
+            flow_run_id = prefect.context.get("flow_run_id")
+
+            query = {
+                "query": {
+                    with_args("flow_run", {"where": {"id": {"_eq": flow_run_id}}}): {
+                        "flow": {"name": True, "storage": True,},
+                    }
+                }
+            }
+
+            client = Client()
+            result = client.graphql(query)
+            flow_run = result.data.flow_run[0]
+
+            flow_data = flow_run.flow
+            storage_schema = prefect.serialization.storage.StorageSchema()
+            storage = storage_schema.load(flow_data.storage)
+
+            ## populate global secrets
+            secrets = prefect.context.get("secrets", {})
+            for secret in storage.secrets:
+                secrets[secret] = prefect.tasks.secrets.PrefectSecret(name=secret).run()
```

Comment:
It looks like this code is duplicated almost verbatim across multiple environments - perhaps this should be a method on the base `Environment` class? Could this `run_flow` implementation be used on all environments? If not, why not?

Reply:
Ah great callout, this can definitely be abstracted out